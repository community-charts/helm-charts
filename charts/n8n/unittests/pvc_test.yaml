suite: test pvc

templates:
  - pvc.yaml

release:
  name: n8n
  namespace: n8n

chart:
  version: 1.0.0
  appVersion: 1.0.0

set:
  main:
    persistence:
      enabled: true
  worker:
    persistence:
      enabled: true

tests:
  - it: should not create pvc resource for main node when main.count is 1 and existingClaim is set
    set:
      main:
        count: 1
        persistence:
          existingClaim: "test-existing-claim"
      worker:
        count: 1
        persistence:
          existingClaim: "test-existing-claim"
    asserts:
      - hasDocuments:
          count: 0

  - it: should not create pvc resource for main node when main.count is bigger than 1 and existingClaim is not set
    set:
      main:
        count: 2
        persistence:
          existingClaim: ""
      worker:
        count: 2
        persistence:
          existingClaim: ""
    asserts:
      - hasDocuments:
          count: 0

  - it: should create pvc resource for main node when main.count is 1 and existingClaim is not set
    set:
      main:
        count: 1
        persistence:
          volumeName: "test-volume"
          existingClaim: ""
          mountPath: "/home/node/.n8n"
          subPath: ""
          storageClass: "test-storage-class"
          accessMode: ReadWriteOnce
          size: 1Gi
          annotations:
            key-annotation: value-annotation
          labels:
            key-label: value-label
    documentSelector:
      path: metadata.name
      value: n8n-main-persistence
    asserts:
      - equal:
          path: metadata.labels.key-label
          value: value-label
      - equal:
          path: metadata.annotations.key-annotation
          value: value-annotation
      - lengthEqual:
          path: spec.accessModes
          count: 1
      - equal:
          path: spec.accessModes[0]
          value: ReadWriteOnce
      - equal:
          path: spec.resources.requests.storage
          value: 1Gi
      - equal:
          path: spec.storageClassName
          value: test-storage-class

  - it: should create pvc resource for worker node when worker.count is 1 and existingClaim is not set
    set:
      worker:
        count: 1
        persistence:
          volumeName: "test-volume"
          existingClaim: ""
          mountPath: "/home/node/.n8n"
          subPath: ""
          storageClass: "test-storage-class"
          accessMode: ReadWriteOnce
          size: 1Gi
          annotations:
            key-annotation: value-annotation
          labels:
            key-label: value-label
    documentSelector:
      path: metadata.name
      value: n8n-worker-persistence
    asserts:
      - equal:
          path: metadata.labels.key-label
          value: value-label
      - equal:
          path: metadata.annotations.key-annotation
          value: value-annotation
      - lengthEqual:
          path: spec.accessModes
          count: 1
      - equal:
          path: spec.accessModes[0]
          value: ReadWriteOnce
      - equal:
          path: spec.resources.requests.storage
          value: 1Gi
      - equal:
          path: spec.storageClassName
          value: test-storage-class

  - it: should match snapshot of main persistence
    set:
      main:
        count: 1
        persistence:
          enabled: true
      worker:
        persistence:
          enabled: false
    asserts:
      - matchSnapshot: { }

  - it: should match snapshot of worker persistence
    set:
      main:
        persistence:
          enabled: false
      worker:
        count: 1
        persistence:
          enabled: true
    asserts:
      - matchSnapshot: { }
