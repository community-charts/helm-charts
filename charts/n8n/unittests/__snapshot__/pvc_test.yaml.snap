should match snapshot of main persistence:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      annotations:
        helm.sh/resource-policy: keep
      labels:
        app.kubernetes.io/component: persistence
        app.kubernetes.io/instance: n8n
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: n8n-main-persistence
        app.kubernetes.io/part-of: n8n
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: n8n-1.0.0
      name: n8n-main-persistence
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 8Gi
should match snapshot of worker persistence:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      annotations:
        helm.sh/resource-policy: keep
      labels:
        app.kubernetes.io/component: persistence
        app.kubernetes.io/instance: n8n
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: n8n-worker-persistence
        app.kubernetes.io/part-of: n8n
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: n8n-1.0.0
      name: n8n-worker-persistence
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 8Gi
